tfe-scripts/test/unit/add-item.bats

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bats
+
+load ../test_helper
+
+setup() {
+  SAMPLE_VARS_DATA="$(cat "${BATS_TEST_DIRNAME}/../fixtures/example_vars.json")"
+  TARGET_KEY="variable_2"
+  NEW_ITEM="slug-env2,app2"
+  EXPECTED="$(echo "[\"slug-env1,app1\",\"slug-env2,app2\"]" | jq ". | @base64")"
+}
+
+teardown() {
+  if [ -z "$TEST_FUNCTION" ]; then
+    shellmock_clean
+  fi
+}
+
+@test "add a new item into json array parsed from a sample Terraform API response data" {
+  data="$(echo "$SAMPLE_VARS_DATA" | jq -r ".data[] | select(.attributes.key == \"$TARGET_KEY\") | .")"
+  list="$(echo "$data" | jq -r ".attributes.value")"
+  result="$(echo "$list" | jq ". + [\"$NEW_ITEM\"] | unique | @base64")"
+  echo "output = $result"
+  [ "$result" == "$EXPECTED" ]
```

Comment:
This test needs to be changed to directly test your script instead of copying the jq query here. However, I don't think that this is something that needs to be done now. Please create a card to schedule this for a future sprint, as there are higher priority items that need to be taken care of.

Reply:
I agree with that, I was thinking of creating a function to process that logic to be used for the scripts and tests, but as I removed much complexity by using jq instead of loop and escaped quotes, I didn't do it somehow.